feat: rename yml file before transferring (on top of #467)

eduardacoppo opened the pull request:
https://app.shortcut.com/tidewise/story/69064/implement-way-to-change-the-name-of-info-yml-on-process-servers

On top of #467

doudou reviewed (commented):
Same comment. Renaming should be done during transfer by log_runtime_archive

jhonasiv commented:
Ah, I approved by accident, but you can consider it an approval after you satisty @doudou comment

doudou commented:
> Ah, I approved by accident, but you can consider it an approval after you satisty @doudou comment

I essentially said "this pull request should be closed" ...

eduardacoppo committed:
fix: optionally upload info.yml under a different name

This is meant to avoid collisions between the different satellite machines

jhonasiv commented on lib/syskit/cli/log_runtime_archive.rb:
```diff
@@ -196,6 +198,17 @@ def self.transfer_dataset(
+                info_file = dataset_path / "info.yml"
+                return unless info_file.file?
+
+                new_info_file =
+                    dataset_path / "info_#{Time.now.strftime('%Y%m%d-%H%M')}.yml"
```
Is this from a talk you had in person with someone? Apart from the source of the info not being obvious in the file name (which I think is not that big a deal when the info is mostly used by other tools), I think you still run the risk of conflict if you stop at the "minutes" in the name. You should at least include the seconds in the name (or ms if you are feeling zealous)

doudou replied:
I have no idea why she did this ...

doudou committed:
chore: style

doudou committed:
feat: add the --info-name option to transfer and watch-transfer

doudou committed:
fix: race condition while waiting for log completion in tests

The remote process manager's log upload state 'pending' count would be very hard
to get race-free. In this case, the pending count it taken (quite) after the
result queue has been emptied, in a code that looks like

    result = empty_queue
    return SomeObject.new(pending.value, result)

If a result is queued between the two lines (which in practice are much more
than two lines), pending will be decremented but result does not contain the
result.

wvmcastro reviewed (approved):
In a redundant system we must remember to use names with ids i.e. motion-box-1, ..., motion-box-n

doudou commented:
> In a redundant system we must remember to use names with ids i.e. motion-box-1, ..., motion-box-n

Good point. Documenting this now (as well as integrating in Ansible).
